--- a/src/rtl8261d_main.c
+++ b/src/rtl8261d_main.c
@@ -1486,7 +1486,7 @@ static struct phy_driver rtl8261_drv[] =
 		.suspend          = rtl8261x_suspend,
 		.resume           = rtl8261x_resume,
 		.read_page        = rtl8261x_read_page,
 		.write_page       = rtl8261x_write_page,
-		.set_loopback     = rtl8261x_set_loopback,
+		.set_loopback     = (int (*)(struct phy_device *, bool, int))rtl8261x_set_loopback,
 	},
 };
