--- a/src/rtl837x_mdio.c
+++ b/src/rtl837x_mdio.c
@@ -12,4 +12,5 @@
 #include <linux/phy.h>
 #include <linux/sfp.h>
+#include <linux/version.h>
 
 #include "rtl837x_gsw.h"
@@ -572,3 +573,5 @@ static void rtl837x_sfp_module_insert(vo
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 5, 0)
 	sfp_parse_support(gsw->sfp_bus, id, support, interfaces);
+#endif
 }
--- a/src/rtl837x_gpio.c
+++ b/src/rtl837x_gpio.c
@@ -79,7 +79,7 @@ static int rtl837x_gpio_get(struct gpio_chip *gc, unsigned int offset)
 	return RTL_READ_GPIO_PIN(gsw, offset) & BIT(offset % 32);
 }
 
-static void rtl837x_gpio_set(struct gpio_chip *gc, unsigned int offset, int value)
+static int rtl837x_gpio_set(struct gpio_chip *gc, unsigned int offset, int value)
 {
 	struct rtl837x_gsw_priv *gsw = gpiochip_get_data(gc);
 
@@ -90,6 +90,7 @@ static void rtl837x_gpio_set(struct gpio_chip *gc, unsigned int offset, int val
 		RTL_CLEAR_GPIO_PIN(gsw, offset);
 	}
 	raw_spin_unlock_irqrestore(&gsw->lock, flags);
+	return 0;
 }
 
 static const struct gpio_chip rtl837x_gp = {
